ocs_ci/ocs/pgsql.py

```diff
@@ -663,3 +664,193 @@ def pgsql_full(self):
         pgbench_pods = self.get_pgbench_pods()
         # Validate pgbench run and parse logs
         self.validate_pgbench_run(pgbench_pods)
+
+    def generate_random_date(self, min_year=1900, max_year=datetime.now().year):
+        """
+        Generate a random date in yyyy-mm-dd format.
+
+        Args:
+            min_year (int) : The minimum year for which the date can be generated.
+            max_year (int) : The maximum year for which the date can be generated.
+
+        Returns:
+            str : Random date in yyyy-mm-dd format.
+
+        """
+        start = datetime(min_year, 1, 1, 00, 00, 00)
+        years = max_year - min_year + 1
+        end = start + timedelta(days=365 * years)
+        return (start + (end - start) * random.random()).strftime("%Y-%m-%d")
+
+    def generate_random_string(self, length):
+        """
+        Generate a random string of given length.
+
+        Args:
+            length (int) : The length of the string that needs to be generated.
+
+        Returns:
+            str : Random string of given length.
+
+        """
+        return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+    def run_pgsql_command(self, postgres_pod, command, select=False):
+        """
+        Run pgsql command on the postgres POD
+
+        Args:
+            postgres_pod (POD) : POD object on which postgres is running.
+            command (str) : The pgsql command that we want to run.
+            select (bool) : True if the command is of type select, False otherwise.
+
+        Returns:
+            str : Output of the select query.
+
+        """
+        out = postgres_pod.exec_cmd_on_pod(f'psql -U postgres -c "{command}" ')
+        if select:
+            ind = out.index("-")
+            result = list()
+            result.append("\n" + out[:ind])
+            result.append("-" * len(result[0]))
+            result.extend(
+                re.findall(r"\d+\s\|\s[A-Z0-9]+\s\|\s\d{4}-\d{2}-\d{2}\s", out)
+            )
+            result.append("-" * len(result[0]))
+            log.info("\n".join(result))
+            return out
+        log.info(out)
+
+    def run_insert_operation(self, num_of_ops, postgres_pod, row_index, table_name):
+        """
+        Insert num_of_ops number of rows in the table using run_pgsql_command.
+
+        Args:
+            num_of_ops (int) : Total number of records we want to be inserted.
+            postgres_pod (POD) : POD object on which postgres is running.
+            row_index (int) : Starting index of the rows to be inserted.
+            table_name (str) : Name of the table.
+
+        """
+        log.info(f"Running {num_of_ops} insert operations.")
+        res = ""
+        for row_num in range(row_index + 1, row_index + num_of_ops + 1):
+            random_username = self.generate_random_string(50)
+            res += f"({row_num}, '{random_username}', '{self.generate_random_date()}'),"
```

Comment:
Is it ok to have comma in the end of the command ?
maybe you should delete the last comma outside the loop.

also I think you should have a space after the comma.

Reply:
the res contains the randomly generated data to insert into the table. Each of the entry will be comma separated. 
Below is the output snip of it,

(1, 'YT00ARUZOC8NESHADPZ3SJSTQ42G4INYN43RZS6XLFHSY1YNX0', '1921-11-28'),(2, 'HRC4EZW0ZZIREEMUV6MIOZN12DJB5WB7CEDEF109QI6G4WXE90', '1906-11-19'),(3, 'ON6YVTDG74GVYCXC7659K9OU8UQMB7T76B24JT18TOP0S8FN3L', '1984-08-25'),